cmd/api/http/http_test.go

```diff
@@ -0,0 +1,308 @@
+
+		is.True(response.Result().StatusCode == 201)
+		is.Equal(string(body), expectedJSONresponse)
+
+	})
+
+	t.Run("expected invalid json error", func(t *testing.T) {
+		is := is.New(t)
+
+		invalidBookToCreate := `{
+				"name": "test with missing coma after price",
+				"price": 100
+				"inventory": 99
+			}`
+		expectedJSONresponse := `{"error_code":102,"error_message":"invalid json request.invalid character '\"' after object key:value pair"}
+`
+
+		request, _ := http.NewRequest(http.MethodPost, "/books", strings.NewReader(invalidBookToCreate))
+		response := httptest.NewRecorder()
+
+		server.Handler.ServeHTTP(response, request)
+
+		body, _ := io.ReadAll(response.Result().Body)
+
+		is.True(response.Result().StatusCode == 400)
+		is.Equal(string(body), expectedJSONresponse)
+
+	})
+
+	t.Run("expected blank fields error", func(t *testing.T) {
+		is := is.New(t)
+
+		invalidBookToCreate := `{
+			"name": "test with missing inventory",
+			"price": 100
+		}`
+		expectedJSONresponse := `{"error_code":100,"error_message":"all the fields - name, price and inventory - must be filled correctly."}
+`
+
+		request, _ := http.NewRequest(http.MethodPost, "/books", strings.NewReader(invalidBookToCreate))
+		response := httptest.NewRecorder()
+
+		server.Handler.ServeHTTP(response, request)
+
+		body, _ := io.ReadAll(response.Result().Body)
+
+		is.True(response.Result().StatusCode == 400)
+		is.Equal(string(body), expectedJSONresponse)
+
+	})
+}
+func TestListBooks(t *testing.T) {
+
+	ctrl := gomock.NewController(t)
+	mockAPI := httpmock.NewMockServiceAPI(ctrl)
+	bookHandler := bookhttp.NewBookHandler(mockAPI)
+
+	server := bookhttp.NewServer(bookhttp.ServerConfig{Port: 8080}, bookHandler)
+
+	// Setting up, creating books to be listed.
+	var testBookslist []book.Book
+	listSize := 30
+	for i := 0; i < listSize; i++ {
+		b := book.Book{
+			ID:        uuid.New(),
+			Name:      fmt.Sprintf("Book number %06v", i),
+			Price:     toPointer(float32((i * 100) + 1)),
+			Inventory: toPointer(100 - i),
+			CreatedAt: time.Now().UTC().Round(time.Millisecond),
+			UpdatedAt: time.Now().UTC().Round(time.Millisecond),
+			Archived:  false,
+		}
+		testBookslist = append(testBookslist, b)
+	}
+
+	t.Run("lists all books with defalult values, without errors", func(t *testing.T) {
+		is := is.New(t)
+
+		// Setting query parameters
+		params := book.ListBooksRequest{
+			Name:          "",
+			MinPrice:      0,
+			MaxPrice:      book.PriceMax,
+			SortBy:        "name",
+			SortDirection: "asc",
+			Archived:      false,
+			Page:          1,
+			PageSize:      10,
+		}
+		url := "/books"
+
+		expectedReturn := book.PagedBooks{
+			PageCurrent: params.Page,
+			PageTotal:   3,
+			PageSize:    params.PageSize,
+			ItemsTotal:  listSize,
+			Results:     testBookslist[0:10],
+		}
+
+		expectedJSONresponse, err := json.Marshal(pagedBooksToResponse(expectedReturn))
+		is.NoErr(err)
+		expectedJSONresponse = append(expectedJSONresponse, 10)
```

Comment:
This is a bit obscure. What does `10` stands for? lets change it to this so we know what we are appending.
```suggestion
		expectedJSONresponse = append(expectedJSONresponse, []byte("\n")...)